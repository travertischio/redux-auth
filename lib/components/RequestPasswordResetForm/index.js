"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactIntl = require("react-intl");

var _immutable = require("redux-form/immutable");

var _reduxForm = _interopRequireDefault(require("react-form-fields/lib/EmailField/reduxForm"));

var _required = require("validators/lib/required");

var _email = require("validators/lib/email");

var _messages = _interopRequireDefault(require("./messages"));

/**
*
* RequestPasswordResetForm
*
*/
function RequestPasswordResetForm(props) {
  var handleSubmit = props.handleSubmit,
      pristine = props.pristine,
      submitting = props.submitting;
  var formatMessage = props.intl.formatMessage;
  var emailLabel = formatMessage(_messages["default"].email);

  var onSubmit = function onSubmit(event) {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    handleSubmit();
  };

  return _react["default"].createElement("form", {
    onSubmit: onSubmit,
    noValidate: true
  }, _react["default"].createElement(_immutable.Field, {
    id: "email",
    name: "email",
    label: emailLabel,
    placeholder: emailLabel,
    validate: [_required.required, _email.email],
    component: _reduxForm["default"]
  }), _react["default"].createElement("div", null, _react["default"].createElement("button", {
    type: "submit",
    disabled: pristine || submitting
  }, "Submit")));
}

var _default = (0, _immutable.reduxForm)({
  form: 'requestPasswordResetForm'
})((0, _reactIntl.injectIntl)(RequestPasswordResetForm));

exports["default"] = _default;