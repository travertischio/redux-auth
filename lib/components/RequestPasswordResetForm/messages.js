'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactIntl = require('react-intl');

exports.default = (0, _reactIntl.defineMessages)({
  email: {
    id: 'app.components.SignInForm.email',
    defaultMessage: 'E-mail'
  }
}); /*
     * RequestPasswordResetForm Messages
     *
     * This contains all the text for the RequestPasswordResetForm component.
     */