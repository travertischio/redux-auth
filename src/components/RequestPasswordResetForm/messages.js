/*
 * RequestPasswordResetForm Messages
 *
 * This contains all the text for the RequestPasswordResetForm component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  email: {
    id: 'app.components.SignInForm.email',
    defaultMessage: 'E-mail',
  },
});
