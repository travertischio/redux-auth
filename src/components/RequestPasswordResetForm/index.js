/**
*
* RequestPasswordResetForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import {
  Field,
  reduxForm,
} from 'redux-form/immutable';
import EmailField from 'react-form-fields/lib/EmailField/reduxForm';
import { required as requiredValidator } from 'validators/lib/required';
import { email as emailValidator } from 'validators/lib/email';
import messages from './messages';

function RequestPasswordResetForm(props) {
  const {
    handleSubmit,
    pristine,
    submitting,
    valid,
  } = props;
  const { formatMessage } = props.intl;
  const emailLabel = formatMessage(messages.email);

  const onSubmit = (event) => {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    if (valid) {
      handleSubmit();
    }
  };

  return (
    <form onSubmit={onSubmit} noValidate>
      <Field
        id="email"
        name="email"
        label={emailLabel}
        placeholder={emailLabel}
        validate={[requiredValidator, emailValidator]}
        component={EmailField}
      />
      <div>
        <button
          type="submit"
          disabled={pristine || submitting}
        >
          Submit
        </button>
      </div>
    </form>
  );
}

RequestPasswordResetForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  valid: PropTypes.bool.isRequired,
  intl: PropTypes.object.isRequired,
};

export default reduxForm({
  form: 'requestPasswordResetForm',
})(injectIntl(RequestPasswordResetForm));
