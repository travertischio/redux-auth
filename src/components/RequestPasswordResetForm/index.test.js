import React from 'react';
import { createComponentWithIntl } from 'react-unit-testing-utils';

import RequestPasswordResetForm from '.';

describe('<RequestPasswordResetForm />', () => {
  it('should render RequestPasswordResetForm', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<RequestPasswordResetForm />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
