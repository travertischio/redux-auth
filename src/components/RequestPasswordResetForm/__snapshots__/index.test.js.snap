// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestPasswordResetForm /> should render RequestPasswordResetForm 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="email"
      >
        E-mail
      </label>
      <input
        className="textfield"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="E-mail"
        type="email"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
